d2/src/main/java/com/linkedin/d2/discovery/event/IndisAnnouncer.java

```diff
@@ -0,0 +1,19 @@
+package com.linkedin.d2.discovery.event;
+
+import com.linkedin.d2.balancer.properties.PartitionData;
+import com.linkedin.d2.balancer.properties.UriProperties;
+import java.util.Map;
+
+/**
+ * This interfaces handles announcing and deannouncing in LinkedIn's Next Generation Service Discovery ([In]Dis).
+ * It will be implemented internally, and we provide a default implementation {@link NoopIndisAnnouncer}
+ * that open-source restli users can use (or provide their own implementation for their own service discovery system)
+ */
+public interface IndisAnnouncer {
+  void start();
+  void stop();
+  void announce(String cluster, String protocol, String host, int port, String path, Map<Integer, PartitionData> partitionDataMap,
+      Map<String, Object> uriSpecificProperties, UriProperties d2UriProperties);
+
+  void deannounce(String cluster, String protocol, String host, int port, String path);
```

Comment:
Yes, exactly. I don't think we have access to these parameters in the `markDown` function, so we can't easily pass them, but in our implementation in container we can use the existing data from cache rather than starting from scratch. 

This is the relevant change in container to enable that: https://linkedin.ghe.com/multiproduct/container/pull/135/commits/0ffefa26f9863c0adac20a24be06a66c98cc7aab